### PR TITLE
Metrics implementation

### PR DESCRIPTION
**Pull request by r4f4ss**
Implementation/fix of the following metrics:

portal/\<NETWORK\>/radius_ratio
portal/\<NETWORK\>/entry_count
portal/\<NETWORK\>/content_storage
portal/\<NETWORK\>/total_storage

### CODE REVIEW

File: portalnetwork/beacon/storage.go [hunk 1]
```diff
@@ -41,6 +53,44 @@ func NewBeaconStorage(config storage.PortalStorageConfig) (storage.ContentStorag
 	if err := bs.setup(); err != nil {
 		return nil, err
 	}
+	if metrics.Enabled {
+		radiusRatio = metrics.NewRegisteredGaugeFloat64("portal/beacon/radius_ratio", nil)
+		radiusRatio.Update(1)
+
+		entriesCount = metrics.NewRegisteredGauge("portal/beacon/entry_count", nil)
+		log.Info("Counting entities in beacon storage for metrics")
```
Comment (GrapeBaBa) [inline portalnetwork/beacon/storage.go:61]: Maybe use `Debug` or `Trace` is better

File: portalnetwork/beacon/storage.go [hunk 2]
```diff
@@ -41,6 +53,44 @@ func NewBeaconStorage(config storage.PortalStorageConfig) (storage.ContentStorag
 	if err := bs.setup(); err != nil {
 		return nil, err
 	}
+	if metrics.Enabled {
+		radiusRatio = metrics.NewRegisteredGaugeFloat64("portal/beacon/radius_ratio", nil)
```
Comment (GrapeBaBa) [inline portalnetwork/beacon/storage.go:57]: Could we extract this logic to reduce duplicate code in three networks

File: portalnetwork/beacon/storage.go (hunk 1 above)
Comment (r4f4ss) [reply portalnetwork/beacon/storage.go]: Chnaged to `Debug`

File: portalnetwork/beacon/storage.go (hunk 2 above)
Comment (r4f4ss) [reply portalnetwork/beacon/storage.go]: Logic moved to function `NewPortalStorageMetrics` in `portal_metrics.go`

**Comment by thinkAfCod** [comment]
I think we can abstract a Metrics interface and add an implementation called `NoMetrics`, where all methods are empty. 

When metrics are not needed, we can use an instance of `NoMetrics`, so we don't have to check `metrics.Enable` every time.

File: portalnetwork/history/storage.go [hunk 3]
```diff
@@ -447,12 +495,24 @@ func (p *ContentStorage) ReclaimSpace() error {
...
+		}
+	}
 	res, err := p.sqliteDB.Exec(deleteOutOfRadiusStmt, p.nodeId[:], radius.Bytes())
 	if err != nil {
 		return err
 	}
-	count, _ := res.RowsAffected()
+	count, err := res.RowsAffected()
 	p.log.Trace("delete items", "count", count)
+	if metrics.Enabled && err != nil {
```
Comment (thinkAfCod) [inline portalnetwork/history/storage.go:512]: The condition probably be `err == nil`

**Comment by r4f4ss** [comment]
> I think we can abstract a Metrics interface and add an implementation called `NoMetrics`, where all methods are empty.
> 
> When metrics are not needed, we can use an instance of `NoMetrics`, so we don't have to check `metrics.Enable` every time.

Definitely will make the code more clear, but I am not sure if a function `call` to the interface is computationally cheapper than an `if` verification. As alternative, what about replace the duplicate code:
```go
if metrics.Enabled {
	portalStorageMetrics.EntriesCount.Dec(count)
}
```
in a function that verify:
```go
func verifyDec(metric){
    if metrics.Enabled {
	    metric.Dec(count)
    }
}
```
and it is more clear in the code to call simple:
```go
verifyDec(metric)
```

File: portalnetwork/history/storage.go (hunk 3 above)
Comment (r4f4ss) [reply portalnetwork/history/storage.go:512]: Yes, thanks, fixed locally, I will commit soon

**Review by fearlessfe — APPROVED** [review]

**Comment by GrapeBaBa** [comment]
> > I think we can abstract a Metrics interface and add an implementation called `NoMetrics`, where all methods are empty.
> > When metrics are not needed, we can use an instance of `NoMetrics`, so we don't have to check `metrics.Enable` every time.
> 
> Definitely will make the code more clear, but I am not sure if a function `call` to the interface is computationally cheapper than an `if` verification. As alternative, what about replace the duplicate code:
> 
> ```go
> if metrics.Enabled {
> 	portalStorageMetrics.EntriesCount.Dec(count)
> }
> ```
> 
> in a function that verify:
> 
> ```go
> func verifyDec(metric){
>     if metrics.Enabled {
> 	    metric.Dec(count)
>     }
> }
> ```
> 
> and it is more clear in the code to call simple:
> 
> ```go
> verifyDec(metric)
> ```
> 
> EDIT: Nevermind, It is the worst solution, uses an function call and an if comparison, higher computationally cost

Just follow the geth style is OK.

**Comment by fearlessfe** [comment]
By the way, the code of storage is according from the early version of the trin.  Now, the trin has update the storage code, maybe we can check there code to decide if we should do some update.

**Review by GrapeBaBa — APPROVED** [review]

**Pull request merged**